addons/account_edi/models/account_edi_document.py

```diff
@@ -203,10 +214,21 @@ def _postprocess_cancel_edi_results(documents, edi_result):
                         attachments_to_unlink |= old_attachment
 
                 elif not move_result.get('success'):
+                    blocking_level = move_result.get('blocking_level', DEFAULT_BLOCKING_LEVEL) if move_result.get('error') else False
                     document.write({
                         'error': move_result.get('error', False),
-                        'blocking_level': move_result.get('blocking_level', DEFAULT_BLOCKING_LEVEL) if move_result.get('error') else False,
+                        'blocking_level': blocking_level,
                     })
+                    title = _('An error occurred during cancelling electronic document')
+                    document.move_id.message_post(
+                        title=title,
+                        body=move_result.get('error', False),
+                    )
+                    # to be sure blocking_level is one of the standard values and otherwise switch to the default one
+                    getattr(_logger, blocking_level or DEFAULT_BLOCKING_LEVEL, getattr(_logger, DEFAULT_BLOCKING_LEVEL))(
```

Comment:
```suggestion
                    blocking_level = document.blocking_level or DEFAULT_BLOCKING_LEVEL
                    log_level = blocking_level.lower() if blocking_level.upper() in logging._nameToLevel.keys() else 'error'
                    getattr(_logger, log_level)(
```
It is safer but still not quite readable